packages/block-library/src/button/block.json

```diff
@@ -52,6 +52,9 @@
 		},
 		"gradient": {
 			"type": "string"
+		},
+		"width": {
+			"type": "number"
```

Comment:
Thinking about possible futures for this block, is this the best data representation? How would we express custom width values without introducing compatibility issues with the current schema? That is, this new `width` attribute is not any `number` but actually a percentage represented as `0 <= n <= 100`. Would it be better if it were a real number _x_ `0 <= x <= 1` to distinguish from full pixel values like `width: 200`? Would it be better if termed `widthPercentage`?

All of these are open questions — I don't have a formed opinion on this.

Reply:
Another possibility here would be to add an additional `widthUnit` attribute if we decide to introduce custom width as pixel values. The `search` block takes that approach.

Reply:
Yes this is how the search block handles this, as well as the cover block. I think this approach is future proof based on that.

Reply:
This is the approach the search and cover blocks take, so this should be future proof for adding a unit in the future.